lib/manager/composer/utils.ts

```diff
@@ -79,6 +83,21 @@ export function getPhpConstraint(constraints: Record<string, string>): string {
   return null;
 }
 
+export function hasComposerDependencyInstalled(
+  lock: ComposerLock,
+  dependency: string,
+  includeDev?: boolean
+): boolean {
+  if (lock.packages?.some((p) => p.name === dependency)) {
+    return true;
+  }
+
+  return (
+    includeDev === true &&
+    lock['packages-dev']?.some((p) => p.name === dependency) === true
+  );
+}
```

Comment:
I think about something like this:

```suggestion

const depRequireInstall = new Set(['symfony/flex']);

export function requireComposerDependencyInstalled(
  lock: ComposerLock
): boolean {
  return (
    lock.packages?.some((p) => depRequireInstall.has(p.name)) === true ||
    lock['packages-dev']?.some((p) => depRequireInstall.has(p.name)) === true
  );
}
```

Reply:
Okay, see latest update.